modules/core/src/smithy4s/schema/CollectionTag.scala

```diff
@@ -100,6 +106,18 @@ object CollectionTag {
       }
   }
 
+  private[this] def copyOf[A](original: Array[A], newLength: Int): Array[A] = ((original: @unchecked) match {
+    case x: Array[AnyRef] => java.util.Arrays.copyOf(x, newLength)
```

Comment:
TIL you can pattern match against specialised array types and erasure is not a problem